network/round-robin.go

```diff
@@ -0,0 +1,23 @@
+package network
+
+import (
+	"sync/atomic"
+)
+
+type RoundRobin struct {
+	proxies []IProxy
+	next    uint32
```

Comment:
You are using atomic the old way

You can use atomic.Int32

Please take a look at the documentation.

https://pkg.go.dev/sync/atomic#Int32

The code is way simpler since it was added

Reply:
Thanks for the review. I’ll check it.